src/analysis/special_functions/log/deriv.lean

```diff
@@ -268,4 +268,57 @@ begin
       by simpa [pow_succ'] using mul_le_of_le_one_right (pow_nonneg (abs_nonneg x) i) (le_of_lt h) }
 end
 
+/-- Power series expansion of `log(1 + x) - log(1 - x)` for `|x| < 1`. -/
+lemma has_sum_log_sub_log_of_abs_lt_1 {x : ℝ} (h : |x| < 1) :
+  has_sum (λ k : ℕ, (2 : ℝ) * (1 / (2 * k + 1)) * x ^ (2 * k + 1)) (log (1 + x) - log(1 - x)) :=
+begin
+  have h₁ := has_sum_pow_div_log_of_abs_lt_1 h,
+  have h₂ := has_sum_pow_div_log_of_abs_lt_1 (eq.trans_lt (abs_neg x) h),
+  replace h₂ := (has_sum_mul_left_iff (λ h : (-1 = (0 : ℝ)), one_ne_zero $ neg_eq_zero.mp h)).mp h₂,
+  rw [neg_one_mul, neg_neg, sub_neg_eq_add 1 x] at h₂,
+  have h₃, from has_sum.add h₂ h₁,
+  rw ←sub_eq_add_neg at h₃,
+  let term := λ n : ℕ, (-1) * ((-x) ^ (n + 1) / ((n : ℝ) + 1)) + x ^ (n + 1) / (n + 1),
+  rw ←function.injective.has_sum_iff (nat.mul_right_injective two_pos) _ at h₃,
+  { suffices h_term_eq_goal :
+    term ∘ (*) 2 = λ k : ℕ, 2 * (1 / (2 * k + 1)) * x ^ (2 * k + 1),
+    { rw h_term_eq_goal at h₃, exact h₃ },
+      apply funext (λ n, _),
+      rw [function.comp_app],
+      dsimp only [term],
+      rw [odd.neg_pow (⟨n, rfl⟩ : odd (2 * n + 1)) x, neg_one_mul, neg_div, neg_neg, nat.cast_mul,
+        nat.cast_two],
+      ring_nf },
```

Comment:
```suggestion
      rw [odd.neg_pow (⟨n, rfl⟩ : odd (2 * n + 1)) x]
      push_cast,
      ring_nf },
```
No fixed rule here, just my personal preference.  Typically, using tactics will make the structure clearer than using a list of explicit rewrites.